File: frontend/src/app/types/event.ts
export type Xevent = {
    id: number;
    name: string;
    flyer: { src: string; alt: string };
    genre: string[];
    subtitle: string;
    description: string;
    location: { city: string; address: string };
    startTime: Date;
    price: number;
    artists: string[];
}

export type ApiEvent = {
  id: number;
  flyer: string;
  name: string;
  genre: string;
  subtitle: string;
  description: string;
  city: string;
  address: string;
  startTime: string;
  price: number;
  artists: { id: number; name: string }[];
  organizers: { id: number; name: string }[];
};

File: backend/src/main/java/com/tickethub/domain/Event.java
package com.tickethub.domain;

import java.math.BigDecimal;
import java.util.Date;
import java.util.Set;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@EqualsAndHashCode(exclude = {"organizers", "artists"})
@ToString(exclude = {"organizers", "artists"})
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "events")
@JsonIdentityInfo(
  generator = ObjectIdGenerators.PropertyGenerator.class, 
  property = "id"
)
public class Event {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String flyer;
    private String name, genre, subtitle, description, city, address;
    private Date startTime;
    private BigDecimal price;
    private int maxAttendees;
    private int avaliablePlaces;
    private int status;

    @ManyToMany
    @JoinTable(
        name = "event_organizer",
        joinColumns = @JoinColumn(name = "event_id"),
        inverseJoinColumns = @JoinColumn(name = "organizer_id")
    )
    private Set<Organizer> organizers;

    @ManyToMany
    @JoinTable(
        name = "event_artist",
        joinColumns = @JoinColumn(name = "event_id"),
        inverseJoinColumns = @JoinColumn(name = "artist_id")
    )
    private Set<Artist> artists;
}
